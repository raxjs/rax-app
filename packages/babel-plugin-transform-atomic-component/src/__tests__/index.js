const { transformSync } = require('@babel/core');
const atomicComponentPlugin = require('../index');

function getTransfromCode(code) {
  return transformSync(code, {
    filename: './',
    presets: [
      [require.resolve('@babel/preset-env'), {
        loose: true,
        modules: false,
        targets: {
          node: '14'
        }
      }],
    ],
    plugins: [
      [require.resolve( '@babel/plugin-syntax-jsx' )],
      [atomicComponentPlugin, {
        useBuiltIns: true,
      }],
    ],
  }).code;
}

describe('transform atomic components', () => {
  // rax-view
  it('tansform rax-view with className', () => {
    expect(
      getTransfromCode( `
        import View from 'rax-view';
        <View className="test"></View>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-view with className expression', () => {
    expect(
      getTransfromCode( `
        import View from 'rax-view';
        <View className={ expr ? 'a' : b }></View>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-view with children', () => {
    expect(
      getTransfromCode( `
        import View from 'rax-view';
        <View className="klass">
          I am JSXText
          <span className="a">hello</span>
          { expr }
        </View>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-view with style', () => {
    expect(
      getTransfromCode( `
        import View from 'rax-view';
        <View className="klass" style={{ border: a ? b : c, color: '#FFF', backgroundColor: expr + 'string' }}>
        </View>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-view with style and other props', () => {
    expect(
      getTransfromCode( `
        import View from 'rax-view';
        <View data-x="a" className="klass" style={{ border: a ? b : c, color: '#FFF', backgroundColor: expr + 'string' }}>
        </View>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-view with innerHTML', () => {
    expect(
      getTransfromCode( `
        import View from 'rax-view';
        <View className="test" dangerouslySetInnerHTML={{__html: "<div>a</div>"}}>
        </View>
      ` )
    ).toMatchSnapshot()
  });

  // rax-text
  it('tansform rax-text with className', () => {
    expect(
      getTransfromCode( `
        import Text from 'rax-text';
        <Text className="test"></Text>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-text with className expression', () => {
    expect(
      getTransfromCode( `
        import Text from 'rax-text';
        <Text className={ expr ? 'a' : b }></Text>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-text with children', () => {
    expect(
      getTransfromCode( `
        import Text from 'rax-text';
        <Text className="klass">
          I am JSXText
          <span className="a">hello</span>
          { expr }
        </Text>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-text with style', () => {
    expect(
      getTransfromCode( `
        import Text from 'rax-text';
        <Text style={{ border: a ? b : c, color: '#FFF', backgroundColor: expr + 'string' }}>
        </Text>
      ` )
    ).toMatchSnapshot()
  });

  it('tansform rax-text with style and other props', () => {
    expect(
      getTransfromCode( `
        import Text from 'rax-text';
        <Text data-x="a" className="klass" style={{ border: a ? b : c, color: '#FFF', backgroundColor: expr + 'string' }}>
        </Text>
      ` )
    ).toMatchSnapshot()
  });
});
