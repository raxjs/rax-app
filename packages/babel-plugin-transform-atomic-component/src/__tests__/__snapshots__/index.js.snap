// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform atomic components tansform rax-text with children 1`] = `
"var _toRaxTextProps = function (props) {
  const {
    className,
    style = {},
    numberOfLines,
    ...rest
  } = props;
  const prefixCls = 'rax-text-v2';
  const lines = typeof numberOfLines === 'string' ? parseInt(numberOfLines, 10) : numberOfLines;
  let classNames = [prefixCls, className];

  if (lines) {
    classNames.push(prefixCls + '--overflow-hidden');

    if (lines === 1) {
      classNames.push(prefixCls + '--singleline');
    } else {
      classNames.push(prefixCls + '--multiline');
    }
  }

  const lineClamp = lines > 1 ? lines : undefined;
  return { ...rest,
    className: classNames.filter(Boolean).join(' '),
    style: { ...style,
      WebkitLineClamp: lineClamp,
      lineClamp: lineClamp
    }
  };
};

import Text from 'rax-text';
<span {..._toRaxTextProps({
  \\"className\\": \\"klass\\"
})}>
          I am JSXText
          [object Object]
          {expr}
        </span>;"
`;

exports[`transform atomic components tansform rax-text with className 1`] = `
"var _toRaxTextProps = function (props) {
  const {
    className,
    style = {},
    numberOfLines,
    ...rest
  } = props;
  const prefixCls = 'rax-text-v2';
  const lines = typeof numberOfLines === 'string' ? parseInt(numberOfLines, 10) : numberOfLines;
  let classNames = [prefixCls, className];

  if (lines) {
    classNames.push(prefixCls + '--overflow-hidden');

    if (lines === 1) {
      classNames.push(prefixCls + '--singleline');
    } else {
      classNames.push(prefixCls + '--multiline');
    }
  }

  const lineClamp = lines > 1 ? lines : undefined;
  return { ...rest,
    className: classNames.filter(Boolean).join(' '),
    style: { ...style,
      WebkitLineClamp: lineClamp,
      lineClamp: lineClamp
    }
  };
};

import Text from 'rax-text';
<span {..._toRaxTextProps({
  \\"className\\": \\"test\\"
})}></span>;"
`;

exports[`transform atomic components tansform rax-text with className expression 1`] = `
"var _toRaxTextProps = function (props) {
  const {
    className,
    style = {},
    numberOfLines,
    ...rest
  } = props;
  const prefixCls = 'rax-text-v2';
  const lines = typeof numberOfLines === 'string' ? parseInt(numberOfLines, 10) : numberOfLines;
  let classNames = [prefixCls, className];

  if (lines) {
    classNames.push(prefixCls + '--overflow-hidden');

    if (lines === 1) {
      classNames.push(prefixCls + '--singleline');
    } else {
      classNames.push(prefixCls + '--multiline');
    }
  }

  const lineClamp = lines > 1 ? lines : undefined;
  return { ...rest,
    className: classNames.filter(Boolean).join(' '),
    style: { ...style,
      WebkitLineClamp: lineClamp,
      lineClamp: lineClamp
    }
  };
};

import Text from 'rax-text';
<span {..._toRaxTextProps({
  \\"className\\": expr ? 'a' : b
})}></span>;"
`;

exports[`transform atomic components tansform rax-text with style 1`] = `
"var _toRaxTextProps = function (props) {
  const {
    className,
    style = {},
    numberOfLines,
    ...rest
  } = props;
  const prefixCls = 'rax-text-v2';
  const lines = typeof numberOfLines === 'string' ? parseInt(numberOfLines, 10) : numberOfLines;
  let classNames = [prefixCls, className];

  if (lines) {
    classNames.push(prefixCls + '--overflow-hidden');

    if (lines === 1) {
      classNames.push(prefixCls + '--singleline');
    } else {
      classNames.push(prefixCls + '--multiline');
    }
  }

  const lineClamp = lines > 1 ? lines : undefined;
  return { ...rest,
    className: classNames.filter(Boolean).join(' '),
    style: { ...style,
      WebkitLineClamp: lineClamp,
      lineClamp: lineClamp
    }
  };
};

import Text from 'rax-text';
<span {..._toRaxTextProps({
  \\"style\\": {
    border: a ? b : c,
    color: '#FFF',
    backgroundColor: expr + 'string'
  }
})}>
        </span>;"
`;

exports[`transform atomic components tansform rax-text with style and other props 1`] = `
"var _toRaxTextProps = function (props) {
  const {
    className,
    style = {},
    numberOfLines,
    ...rest
  } = props;
  const prefixCls = 'rax-text-v2';
  const lines = typeof numberOfLines === 'string' ? parseInt(numberOfLines, 10) : numberOfLines;
  let classNames = [prefixCls, className];

  if (lines) {
    classNames.push(prefixCls + '--overflow-hidden');

    if (lines === 1) {
      classNames.push(prefixCls + '--singleline');
    } else {
      classNames.push(prefixCls + '--multiline');
    }
  }

  const lineClamp = lines > 1 ? lines : undefined;
  return { ...rest,
    className: classNames.filter(Boolean).join(' '),
    style: { ...style,
      WebkitLineClamp: lineClamp,
      lineClamp: lineClamp
    }
  };
};

import Text from 'rax-text';
<span {..._toRaxTextProps({
  \\"className\\": \\"klass\\",
  \\"style\\": {
    border: a ? b : c,
    color: '#FFF',
    backgroundColor: expr + 'string'
  }
})} data-x=\\"a\\">
        </span>;"
`;

exports[`transform atomic components tansform rax-view with children 1`] = `
"import View from 'rax-view';
<div className=\\"rax-view-v2 klass\\">
          I am JSXText
          <span className=\\"a\\">hello</span>
          {expr}
        </div>;"
`;

exports[`transform atomic components tansform rax-view with className 1`] = `
"import View from 'rax-view';
<div className=\\"rax-view-v2 test\\"></div>;"
`;

exports[`transform atomic components tansform rax-view with className expression 1`] = `
"import View from 'rax-view';
<div className={\\"rax-view-v2 \\" + (expr ? 'a' : b)}></div>;"
`;

exports[`transform atomic components tansform rax-view with innerHTML 1`] = `
"import View from 'rax-view';
<div className=\\"rax-view-v2 test\\" dangerouslySetInnerHTML={{
  __html: \\"<div>a</div>\\"
}}>
        </div>;"
`;

exports[`transform atomic components tansform rax-view with style 1`] = `
"import View from 'rax-view';
<div className=\\"rax-view-v2 klass\\" style={{
  border: a ? b : c,
  color: '#FFF',
  backgroundColor: expr + 'string'
}}>
        </div>;"
`;

exports[`transform atomic components tansform rax-view with style and other props 1`] = `
"import View from 'rax-view';
<div data-x=\\"a\\" className=\\"rax-view-v2 klass\\" style={{
  border: a ? b : c,
  color: '#FFF',
  backgroundColor: expr + 'string'
}}>
        </div>;"
`;
